content/blog/registry-writing-i2c-component/index.md

```diff
@@ -0,0 +1,742 @@
+- Probe the I2C bus and check the sensor presence.
+- Get the sensor ID.
+- Start a new task to read the sensor every 1000 ms.
+
+On the `app_main` function in the `shtc3_read.c` file, add the bus initialization and the device creation functions.
+
+```c
+i2c_master_bus_handle_t i2c_bus_init(uint8_t sda_io, uint8_t scl_io)
+{
+    i2c_master_bus_config_t i2c_bus_config = {
+        .i2c_port = CONFIG_SHTC3_I2C_NUM,
+        .sda_io_num = sda_io,
+        .scl_io_num = scl_io,
+        .clk_source = I2C_CLK_SRC_DEFAULT,
+        .glitch_ignore_cnt = 7,
+        .flags.enable_internal_pullup = true,
+    };
+
+    i2c_master_bus_handle_t bus_handle;
+    ESP_ERROR_CHECK(i2c_new_master_bus(&i2c_bus_config, &bus_handle));
+    ESP_LOGI(TAG, "I2C master bus created");
+    return bus_handle;
+}
+```
+
+The bus initialization is done by the function `i2c_new_master_bus` and the `bus_handle` will be used to create the device on this bus. If you have more devices, you can add to the same bus, but this will be not covered on this article.
+
+```c
+i2c_master_bus_handle_t bus_handle = i2c_bus_init(SHTC3_SDA_GPIO, SHTC3_SCL_GPIO);
+shtc3_handle = shtc3_device_create(bus_handle, SHTC3_I2C_ADDR, CONFIG_SHTC3_I2C_CLK_SPEED_HZ);
+```
+
+Make sure to import the component header file.
+
+```c
+#include "shtc3.h"
+```
+
+Now to proof that the sensor is present in the I2C bus, we can probe and check it before the read and write operations. This probe has the timeout set to 200 ms.
+
+```c
+esp_err_t err = i2c_master_probe(bus_handle, SHTC3_I2C_ADDR, 200);
+```
+
+With the probe result, we can decide if the read task will be created or not.
+
+```c
+if(err == ESP_OK) {
+        ESP_LOGI(TAG, "SHTC3 sensor found");
+        uint8_t sensor_id[2];
+        err = shtc3_get_id(shtc3_handle, sensor_id);
+        ESP_LOGI(TAG, "Sensor ID: 0x%02x%02x", sensor_id[0], sensor_id[1]);
+
+        if(err == ESP_OK) {
+            ESP_LOGI(TAG, "SHTC3 ID read successfully");
+            xTaskCreate(shtc3_read_task, "shtc3_read_task", 4096, NULL, 5, NULL);
+        } else {
+            ESP_LOGE(TAG, "Failed to read SHTC3 ID");
+        }
+
+    } else {
+        ESP_LOGE(TAG, "SHTC3 sensor not found");
+        shtc3_device_delete(shtc3_handle);
+    }
+```
+
+Task to read the sensor.
+
+```c
+void shtc3_read_task(void *pvParameters)
+{
+    float temperature, humidity;
+    esp_err_t err = ESP_OK;
+    shtc3_register_rw_t reg = SHTC3_REG_T_CSE_NM;
+
+    while (1) {
+        err = shtc3_get_th(shtc3_handle, reg, &temperature, &humidity);
+        if(err != ESP_OK) {
+            ESP_LOGE(TAG, "Failed to read data from SHTC3 sensor");
+        } else {
+            ESP_LOGI(TAG, "Temperature: %.2f C, Humidity: %.2f %%", temperature, humidity);
+        }
+        vTaskDelay(1000 / portTICK_PERIOD_MS);
+    }
+}
+```
+
+On this task, the sensor will be woken up, the readout will be read and processed, and then the sensor will return to sleep mode.
+
+#### Bonus

```

Comment:
I would recommend to rename the chapter from bonus to Creating Kconfig (optional).
Many people do not know about the Kconfig and how to use it. This might be good source for a new article.